### PR TITLE
FLPATH-9: Add unit tests for `workflow-examples` package

### PR DESCRIPTION
**Pull request by gciavarrini**
This PR adds unit tests to workflow-example package.

Each example workflow java class has a coverage >= 80%.

image

However, Configuration classes `SimpleWorkflowconfiguration` and `ComplexWorkFlowconfiguration` have **not** unit tests because they are supposed to be tested via integration tests.
That's why this PR excludes classes named `*Configuration.java` from coverage report.

The obtained coverage is:
image

### CODE REVIEW

**Review by RichardW98 — APPROVED** [review]

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/task/BaseWorkFlowTask.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ public void setBeanName(String name) {
 		this.name = name;
 	}
 
-	protected String getParameterValue(WorkContext workContext, String parameterName) throws MissingParameterException {
+	public String getParameterValue(WorkContext workContext, String parameterName) throws MissingParameterException {
```
Comment (pkliczewski) [inline parodos-model-api/src/main/java/com/redhat/parodos/workflow/task/BaseWorkFlowTask.java:50]: Is there any reason to change visibility for this method?

File: workflow-examples/pom.xml [hunk 2]
```diff
@@ -42,13 +46,37 @@
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
-            <version>1.7.36</version>
+            <version>${slf4j.version}</version>
         </dependency>
         <dependency>
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
             <version>1.18.24</version>
```
Comment (pkliczewski) [inline workflow-examples/pom.xml:54]: Can you please move lombok version to properties as well?

File: workflow-examples/src/main/java/com/redhat/parodos/examples/continued/complex/OnboardingAssessmentTask.java [hunk 3]
```diff
@@ -37,7 +37,7 @@
  */
 public class OnboardingAssessmentTask extends BaseAssessmentTask {
 
-	private static final String INPUT = "INPUT";
+	static final String INPUT = "INPUT";
```
Comment (pkliczewski) [inline workflow-examples/src/main/java/com/redhat/parodos/examples/continued/complex/OnboardingAssessmentTask.java:40]: Any reason to change visibility for this constant?

File: workflow-examples/src/main/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTask.java [hunk 4]
```diff
@@ -41,7 +40,7 @@
 @Slf4j
 public class RestAPIWorkFlowTask extends BaseInfrastructureWorkFlowTask {
 
-	private static final String PAYLOAD_PASSED_IN_FROM_SERVICE = "PAYLOAD_PASSED_IN_FROM_SERVICE";
+	static final String PAYLOAD_PASSED_IN_FROM_SERVICE = "PAYLOAD_PASSED_IN_FROM_SERVICE";
```
Comment (pkliczewski) [inline workflow-examples/src/main/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTask.java:43]: same here?

File: workflow-examples/src/main/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTask.java [hunk 5]
```diff
@@ -68,11 +66,15 @@ public WorkReport execute(WorkContext workContext) {
 		}
 		catch (Exception e) {
 			log.error("There was an issue with the REST call: {}", e.getMessage());
-
 		}
 		return new DefaultWorkReport(WorkStatus.FAILED, workContext);
 	}
 
+	ResponseEntity<String> executePost(String urlString, String payload) {
```
Comment (pkliczewski) [inline workflow-examples/src/main/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTask.java:73]: Is there any utility class we can move this method to?

File: workflow-examples/src/main/java/com/redhat/parodos/examples/simple/SecureAPIGetTestTask.java [hunk 6]
```diff
@@ -75,14 +73,26 @@ public WorkReport execute(WorkContext workContext) {
 		return new DefaultWorkReport(WorkStatus.FAILED, workContext);
 	}
 
+	ResponseEntity<String> restExchange(String urlString, String username, String password) {
```
Comment (pkliczewski) [inline workflow-examples/src/main/java/com/redhat/parodos/examples/simple/SecureAPIGetTestTask.java:76]: Is there any utility class we can move the methods to?

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/task/BaseWorkFlowTask.java (hunk 1 above)
Comment (gciavarrini) [reply parodos-model-api/src/main/java/com/redhat/parodos/workflow/task/BaseWorkFlowTask.java:50]: Yes, I need to invoke `getParameterValue()` method in [RestAPIWorkFlowTaskTest](https://github.com/redhat-developer/parodos/blob/92236853d9327c02a3ed096032bcf743c9062e8e/workflow-examples/src/test/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTaskTest.java#L44) and [SecureAPIGetTestTaskTest](https://github.com/redhat-developer/parodos/blob/92236853d9327c02a3ed096032bcf743c9062e8e/workflow-examples/src/test/java/com/redhat/parodos/examples/simple/SecureAPIGetTestTaskTest.java#L53) so `protected` is not enough anymore.
Does make it sense to you?

File: workflow-examples/src/main/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTask.java (hunk 4 above)
Comment (gciavarrini) [reply workflow-examples/src/main/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTask.java:43]: Similar as before, I use this constant in [RestAPIWorkFlowTaskTest](https://github.com/redhat-developer/parodos/blob/92236853d9327c02a3ed096032bcf743c9062e8e/workflow-examples/src/test/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTaskTest.java#L100). I can use another private string constant in the test if you think it's better.

File: parodos-model-api/src/main/java/com/redhat/parodos/workflow/task/BaseWorkFlowTask.java (hunk 1 above)
Comment (pkliczewski) [reply parodos-model-api/src/main/java/com/redhat/parodos/workflow/task/BaseWorkFlowTask.java:50]: Oh I see that those tests are in different package.

File: workflow-examples/src/main/java/com/redhat/parodos/examples/continued/complex/OnboardingAssessmentTask.java (hunk 3 above)
Comment (pkliczewski) [reply workflow-examples/src/main/java/com/redhat/parodos/examples/continued/complex/OnboardingAssessmentTask.java:40]: OK

File: workflow-examples/src/main/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTask.java (hunk 4 above)
Comment (pkliczewski) [reply workflow-examples/src/main/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTask.java:43]: ack

File: coverage/pom.xml [hunk 7]
```diff
@@ -57,6 +57,11 @@
         <groupId>org.jacoco</groupId>
         <artifactId>jacoco-maven-plugin</artifactId>
         <version>0.8.8</version>
+        <configuration>
```
Comment (eloycoto) [inline coverage/pom.xml:60]: Here u need to add workflow-examples as a dependency, no?

File: workflow-examples/src/main/java/com/redhat/parodos/examples/continued/complex/OnboardingAssessmentTask.java (hunk 3 above)
Comment (eloycoto) [reply workflow-examples/src/main/java/com/redhat/parodos/examples/continued/complex/OnboardingAssessmentTask.java:40]: I wouldn't change this at all. If you change this on the code, there is noway to identify that it's a big change affecting the thing, I'll define in the test

**Review by eloycoto — APPROVED** [review]

File: coverage/pom.xml (hunk 7 above)
Comment (gciavarrini) [reply coverage/pom.xml:65]: You're right! I forgot to push the change: thanks!

File: workflow-examples/src/main/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTask.java (hunk 4 above)
Comment (gciavarrini) [reply workflow-examples/src/main/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTask.java]: Used a private string constant in the test.

File: workflow-examples/src/main/java/com/redhat/parodos/examples/continued/complex/OnboardingAssessmentTask.java (hunk 3 above)
Comment (gciavarrini) [reply workflow-examples/src/main/java/com/redhat/parodos/examples/continued/complex/OnboardingAssessmentTask.java]: @eloycoto you're right, changed.

File: workflow-examples/src/main/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTask.java (hunk 5 above)
Comment (pkliczewski) [reply workflow-examples/src/main/java/com/redhat/parodos/examples/simple/RestAPIWorkFlowTask.java]: Great, thank you!

**Review by pkliczewski — COMMENTED** [review]
/lgtm

**Pull request merged**